### PR TITLE
Sidebar Navigation Boilerplate

### PR DESCRIPTION
**Pull request by camckin10**
This PR branches off of Innocent Tech [custom Tailwind config](https://github.com/softwareforgood/innocent_tech_app/pull/9)

This PR is tests the custom Tailwind configurations through the creation of the sidebar navigation boilerplate. 

### Changes
Give a short summary of the changes made in this PR. 

- [x] Separated 'views' folder and 'layout' folder
- [x] Created sidebar_nav boilerplate layout
- [x] Deleted test student model and controller code 
- [x] Render link of sidebar_nav layout to application.html.erb
- [x] Uploaded test Innocent Tech logo 

### Link to Issue 
Link Asana ticket here!
[Subtask #3 of Establish Basic UI Layouts](https://app.asana.com/0/338012126614790/1206547209040176/f)

### Comments, Notes, or  Questions
This section is if you want to share something related to the PR with the reviewer. 

### Related Screenshots 
*For FE tickets specifcially, please include a screenshot of your work!*

Screen Shot 2024-02-22 at 4 15 20 PM

### Before you submit for review, Did you...
- [N/A] Added relevant tests (if applicable)
- [x] Linked Asana ticket to PR
- [x] Added comments, notes, or questions for reviewer
- [x] Added relevant screenshots
- [x] Ensured ticket title is clear and concise(i.e. 'Donation Form Feature' instead of 'sfg/chels/donation-form'

### CODE REVIEW

File: app/views/application/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html>
+	<html>
```
Comment (edithemmings) [inline app/views/application/index.html.erb:2]: Is this redundant with what you have in /layouts/application.html.erb? We might not need this file/folder?

File: app/views/layouts/_sidebar_nav.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<!--Sidebar Navigation-->
+<aside id="separator-sidebar" class="fixed left-0 top-0 z-40 h-screen w-64 -translate-x-full transition-transform sm:translate-x-0" aria-label="Sidebar">
+				<div class="h-full overflow-y-auto bg-it-gray-50 px-3 py-4 dark:bg-gray-800">
```
Comment (edithemmings) [inline app/views/layouts/_sidebar_nav.html.erb:3]: The tabbing looks a little off here

File: config/routes.rb [hunk 3]
```diff
@@ -5,9 +5,15 @@
...
+  #get 'up' => 'rails/health#show', as: :rails_health_check
 
   # Defines the root path route ("/")
   # root "posts#index"
   root 'application#index'
+
+  get "application/index", to: "application#index"
+  
+  get "/students", to: "students#index"
+  get "/students/:id", to: "students#show"
```
Comment (edithemmings) [inline config/routes.rb:17]: These two lines can be consolidated into one: `resources :students, only: %i[index new]`

**Review by edithemmings — COMMENTED** [review]
Left a few comments. It looks like you created some student-related stuff. You'll have some conflicts there now that the initial models were merged. Also, I didn't see a related controller. Is the student index view rendering for you?

There's also a few files related to github actions. Did you mean for those to be in this PR? I didn't look closely at those yet, but I can if you're hoping to get something merged there.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+def index;end
```
Comment (edithemmings) [inline app/controllers/application_controller.rb:4]: You probably want to run rubocop if you haven't already

File: app/views/application/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Comment (edithemmings) [inline app/views/application/index.html.erb:1]: What purpose is this file serving?

Comment (edithemmings) [reply app/views/application/index.html.erb:1]: Oh I might have had it backwards.. one sec

File: config/routes.rb [hunk 6]
```diff
@@ -5,9 +5,11 @@
...
   # Reveal health status on /up that returns 200 if the app boots with no exceptions, otherwise 500.
   # Can be used by load balancers and uptime monitors to verify that the app is live.
-  get 'up' => 'rails/health#show', as: :rails_health_check
+  # get 'up' => 'rails/health#show', as: :rails_health_check
 
   # Defines the root path route ("/")
   # root "posts#index"
   root 'application#index'
+
+  resources :students, only: %i[index new]
```
Comment (edithemmings) [inline config/routes.rb:14]: Did you mean to leave this in? It's probably fine either way since it will get updated soon, but just wanted to check

Comment (camckin10) [reply config/routes.rb:14]: Nope! Will remove. :)

File: app/assets/stylesheets/application.tailwind.css [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
-  .btn-primary {
-    @apply py-2 px-4 bg-blue-200;
+
+  .nav-tab {
+    @apply flex items-center rounded-lg p-2 text-gray-900 hover:bg-gray-100 dark:text-white dark:hover:bg-gray-700;
   }
-}
 
-*/
+  .tab-title{
```
Comment (edithemmings) [inline app/assets/stylesheets/application.tailwind.css:17]:
```suggestion
  .tab-title {
```

File: app/views/application/index.html.erb (hunk 5 above)
Comment (edithemmings) [reply app/views/application/index.html.erb:1]: Now that you've added the `yield` to the application file, this index is getting rendered within the yield. So when you keep these html, head, body tags, you are nesting them inside the body of the application root document. 

This is why this file should **_only_** contain the content of the app that changes when you go to different views. So in this case:
```
<h1 class="text-3xl text-center font-bold underline">
  Innocent Tech Application
</h1>
```

File: app/views/layouts/_sidebar_nav.html.erb [hunk 8]
```diff
@@ -0,0 +1,46 @@
+<!--Sidebar Navigation-->
+<aside id="separator-sidebar" class="fixed left-0 top-0 z-40 h-screen w-64 -translate-x-full transition-transform sm:translate-x-0" aria-label="Sidebar">
+  <div class="h-full overflow-y-auto bg-it-gray-50 px-3 py-4 dark:bg-gray-800">
+    <ul class="space-y-2 font-medium">
+      <li>
+        <!--innocent_tech logo here-->
+        <%# <img src="./assets/images/it_logo.png" alt="innocent_tech_logo" /> %>
+      </li>
+      <li>
+        <a href="#" class="nav-tab group">
```
Comment (edithemmings) [inline app/views/layouts/_sidebar_nav.html.erb:10]: What is the group class doing here? I see [this documentation](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state), but its not clear to me how it works on its own without the modifiers.

**Review by edithemmings — COMMENTED** [review]
Added some explanation to one of my previous comments + a few nits/questions

File: app/views/layouts/_sidebar_nav.html.erb (hunk 8 above)
Comment (camckin10) [reply app/views/layouts/_sidebar_nav.html.erb:10]: That was a typo I forgot to remove. Initially, I was using the group class.

**Review by edithemmings — APPROVED** [review]
Looks great!

**Pull request merged**